lib/net.js

```diff
@@ -319,14 +319,25 @@ Socket.prototype._onTimeout = function() {
 
 
 Socket.prototype.setNoDelay = function(enable) {
+  if (!this._handle) {
+    this.once('connect',
```

Comment:
Yea, I was under the impression that we would be guaranteed that they would be called in order in which they were added...